panel_view.py

```diff
@@ -309,19 +309,22 @@ def fill_panel(window, update=False, **panel_filter):
 
 # logic for updating panel selection
 
-def get_next_panel_line(line, errors):
+def get_next_panel_line(line, col, errors):
     """Get panel line for next error in buffer.
 
     If not found this means the current line is past last error's buffer line.
     In that case return last error's panel line incremented by one, which will
     place panel selection in empty space between buffer sections.
     """
-    for error in errors:
-        if error["line"] > line:
+    for error in sort_errors(errors):
+        if error["line"] == line and col < error["start"]:
             panel_line = error["panel_line"]
             break
-    else:
-        panel_line = errors[-1]["panel_line"] + 1
+        elif error["line"] > line:
+            panel_line = error["panel_line"]
+            break
+        else:
```

Comment:
This else block belonged to the for loop. I don't understand why it shouldn't anymore.

Other than that, this should do the trick.

Reply:
Oh, you’re right, I messed that up. Should effectively have the result, but it does belong to the for loop.